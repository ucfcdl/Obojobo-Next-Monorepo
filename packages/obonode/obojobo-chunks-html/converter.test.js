import Converter from './converter'

describe('HTML Converter', () => {
	test('slateToObo converts a Slate node to an OboNode with content', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => {
					return {}
				}
			},
			text: 'mockText'
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('slateToObo converts a Slate node to an OboNode with triggers', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => {
					return { triggers: 'mock-triggers' }
				}
			},
			text: 'mockText'
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {}
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node with triggers', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: { triggers: 'mock-triggers' }
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})
})
