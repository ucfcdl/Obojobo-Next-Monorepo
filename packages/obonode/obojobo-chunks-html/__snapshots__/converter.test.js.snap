// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML Converter oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "data": Object {
    "content": Object {
      "triggers": undefined,
    },
  },
  "key": "mockKey",
  "nodes": Array [
    Object {
      "leaves": Array [
        Object {
          "text": undefined,
        },
      ],
      "object": "text",
    },
  ],
  "object": "block",
  "type": "mockType",
}
`;

exports[`HTML Converter oboToSlate converts an OboNode to a Slate node with triggers 1`] = `
Object {
  "data": Object {
    "content": Object {
      "triggers": "mock-triggers",
    },
  },
  "key": "mockKey",
  "nodes": Array [
    Object {
      "leaves": Array [
        Object {
          "text": undefined,
        },
      ],
      "object": "text",
    },
  ],
  "object": "block",
  "type": "mockType",
}
`;

exports[`HTML Converter slateToObo converts a Slate node to an OboNode with content 1`] = `
Object {
  "children": Array [],
  "content": Object {
    "html": "mockText",
  },
  "id": "mockKey",
  "type": "mockType",
}
`;

exports[`HTML Converter slateToObo converts a Slate node to an OboNode with triggers 1`] = `
Object {
  "children": Array [],
  "content": Object {
    "html": "mockText",
    "triggers": "mock-triggers",
  },
  "id": "mockKey",
  "type": "mockType",
}
`;
